go/datas/commit.go

```diff
@@ -66,6 +68,50 @@ func CommitDescendsFrom(commit types.Struct, ancestor types.Ref, vr types.ValueR
 	return true
 }
 
+// FindCommonAncestor returns the most recent common ancestor of c1 and c2, if
+// one exists, setting ok to true. If there is no common ancestor, ok is set to false.
+func FindCommonAncestor(c1, c2 types.Struct, vr types.ValueReader) (a types.Struct, ok bool) {
+	c1Q, c2Q := &types.RefByHeight{types.NewRef(c1)}, &types.RefByHeight{types.NewRef(c2)}
+
+	makeSet := func(refs types.RefSlice) types.Set {
+		vals := make(types.ValueSlice, refs.Len())
+		for i, r := range refs {
+			vals[i] = r
+		}
+		return types.NewSet(vals...)
```

Comment:
As said IRL: try using TargetHash().String() of these refs and put in a Go map, not types.Set.